src/OrchardCore.Modules/OrchardCore.ContentLocalization/Controllers/ContentCulturePickerController.cs

```diff
@@ -34,16 +34,18 @@ public class ContentCulturePickerController : Controller, IUpdateModel
         }
 
         [HttpGet]
-        public async Task<IActionResult> RedirectToLocalizedContent(string targetCulture, PathString contentItemUrl, QueryString queryString)
+        public async Task<IActionResult> RedirectToLocalizedContent(string targetCulture, PathString contentItemUrl, string queryStringValue)
         {
             targetCulture ??= CultureInfo.InvariantCulture.Name;
 
-            if (String.IsNullOrEmpty(contentItemUrl))
+            if (!contentItemUrl.HasValue)
             {
                 contentItemUrl = "/";
             }
 
-            var url = "~" + contentItemUrl.Add(queryString);
+            var queryString = new QueryString(queryStringValue);
```

Comment:
Why we need to create a new instance again and again, while we can pass the query string as `string` or `PathString`?!!

Reply:
Now we path a string from the CCP / filter (the underlying string of the original QueryString). Then yes we could just use this string because, as said in my comments, here the underlying string is escaped (not the case for a PathString). But using a `QueryString` allows to use its implicit `ToUriComponent()` that does an additional escape for the `#` char, this is what they do in the aspnetcore code.

But as said we can't define the queryString action parameter as a `QueryString`, it throws an exception, so here it is a string, that's why i do it explicitly in the code. For the contentItemUrl this is the binder that is able to bluid for us the `PathString`.

So, for the queryString we bluid a `QueryString` only once and for the above reason.

See my last comments in the thread.